src/storage/base-storage.test.ts

```
@@ -0,0 +1,274 @@
+      const storage = new BrowserStorage('localStorage', 'custom.');
+      storage.set('test', 'value');
+
+      expect(mockStorage.setItem).toHaveBeenCalledWith('custom.test', '"value"');
+
+      window.localStorage = originalLocalStorage;
+    });
+
+    it('should handle non-browser environment', () => {
+      const originalWindow = global.window;
+      // @ts-expect-error - Deleting window for testing non-browser environment
+      delete global.window;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)